code/controllers/subsystem/xenoarchaeology.dm

```diff
@@ -6,41 +6,34 @@ SUBSYSTEM_DEF(xenoarchaeology)
 	///Which console is the main character
 	var/obj/machinery/computer/xenoarchaeology_console/main_console
 
-	///Traits types, referenced for generation
-	var/list/xenoa_activators
-	var/list/xenoa_minors
-	var/list/xenoa_majors
-	var/list/xenoa_malfunctions
+	//All the traits - before we sort them, semi needed for generation
 	var/list/xenoa_all_traits
```

Comment:
instead of doing `T != /datum/xenoartifact_trait/minor` to filter parent path in `compile_artifact_whitelist()`, I think this `all trait variable` shouldn't have these parent paths.